pkg/spdk/engine.go

```diff
@@ -1235,7 +1235,7 @@ func (e *Engine) ReplicaAdd(spdkClient *spdkclient.Client, dstReplicaName, dstRe
 	opts := &api.SnapshotOptions{
 		Timestamp: util.Now(),
 	}
-	updateRequired, replicasErr, engineErr := e.snapshotOperationWithoutLock(spdkClient, replicaClients, snapshotName, SnapshotOperationCreate, opts)
+	updateRequired, replicasErr, engineErr := e.snapshotOperationWithoutLock(spdkClient, replicaClients, snapshotName, SnapshotOperationCreate, opts, nil)
```

Comment:
Not need to pass `labels`?

Reply:
It is a rebuilding snapshot, not a user created one, so there shouldn't be labels to pass.

Reply:
Does replica rebuilding also restore the labels from the source replica to the destination replica?

Reply:
No, because actually to retrieve an xattr from SPDK you have to know xattr's name. This is the reason why in [Lvol struct](https://github.com/longhorn/longhorn-spdk-engine/blob/9125fc466faf348115d95064adba6018f615bad6/pkg/api/types.go#L33) there aren't an array for storing general labels, we have only `SnapshotTimestamp` and `UserCreated`, whicha are stored as xattr but they have a fixed and so known name.

Reply:
Probably some extra development are needed, if inside Longhorn  the only place where we store information is SPDK, we should need a method to get all the xattrs of a lvol.

Reply:
So, does it mean it is possible the labels are all missing after multiple rebuilding?
If Yes, please go create a BUG ticket for fixing the bug.

Reply:
> I mean that general labels will be missing also in the original snapshot, the function `BdevLvolGetXattr` is used only for labels with fixed names. So extra development is needed. I think we can track it in the issue I am working on: [longhorn/longhorn#9808](https://github.com/longhorn/longhorn/issues/9808)

Sure, but remember to finish the replica rebuilding APIs first, because it will impact Shuo's integration progress. You can continue the remaining this after implementing those APIs.